keps/sig-apps/20190305-add-progress-deadline-seconds-to-job.md

```diff
@@ -0,0 +1,132 @@
+---
+title: Add ProgressDeadlineSeconds to Job
+authors:
+  - "@goodluckbot"
+owning-sig: sig-apps
+participating-sigs:
+  - sig-api-machinery
+reviewers:
+  - "@soltysh"
+  - "@erictune"
+  - "@janetkuo"
+approvers:
+  - "@soltysh"
+editor: TBD
+creation-date: 2019-03-05
+last-updated: 2019-03-05
+status: implementable
+see-also:
+  - n/a
+replaces:
+  - n/a
+superseded-by:
+  - n/a
+---
+
+
+# Add ProgressDeadlineSeconds to Job
+
+## Table of Contents
+
+   * [TTL After Finished Controller](#ttl-after-finished-controller)
+      * [Table of Contents](#table-of-contents)
+      * [Summary](#summary)
+      * [Motivation](#motivation)
+         * [Goals](#goals)
+         * [Non-Goals](#non-goals)
+      * [Proposal](#proposal)
+         * [User Stories](#user-stories)
+         * [Detailed Design](#detailed-design)
+            * [Feature Gate](#feature-gate)
+            * [API Object](#api-object)
+         * [Implementation Details/Notes/Constraints](#implementation-details/notes/constraints)
+            * [Implementation Details](#Implementation Details)
+            * [Validation](#Validation)
+         * [Risks and Mitigations](#risks-and-mitigations)
+      * [Graduation Criteria](#graduation-criteria)
+      * [Implementation History](#implementation-history)
+
+
+## Summary
+
+We propose to add an optinal field `ProgressDeadlineSeconds` to `job`. Like `deployment`'s `ProgressDeadlineSeconds`, if this field is set, it is used to fail a `job` if the `job` does not start within the given time, i.e, the progress deadline.
+
+## Motivation
+
+In [#48075][] and [#51153][] we've introduced backoff policy for job, which does not cover certain issues with erroneous jobs. 
+For example, we create a job with miss-typed pull spec, and will find out about it only when the job controller kicks a pod and the pod will result in ImagePullBackOff state. The job will remain active.
```

Comment:
pull spec --> pod spec

nit: the pod is in `waiting` state with `ImagePullBackOff` reason.